homeassistant/components/simplisafe/__init__.py

```diff
@@ -283,30 +299,155 @@ def v3_only(coro):
     return True
 
 
+@dataclass(frozen=True)
+class SimpliSafeWebsocketEvent:
+    """Define a representation of a parsed websocket event."""
+
+    event_data: dict
+
+    changed_by: Optional[str] = field(init=False)
+    event_type: Optional[str] = field(init=False)
+    info: str = field(init=False)
+    sensor_name: str = field(init=False)
+    sensor_type: EntityTypes = field(init=False)
+    timestamp: datetime = field(init=False)
+
+    def __post_init__(self):
+        """Initialize."""
+        object.__setattr__(self, "changed_by", self.event_data["pinName"])
+        object.__setattr__(
+            self, "event_type", get_event_type_from_payload(self.event_data)
+        )
+        object.__setattr__(self, "info", self.event_data["info"])
+        object.__setattr__(self, "sensor_name", self.event_data["sensorName"])
+        object.__setattr__(
+            self, "timestamp", utc_from_timestamp(self.event_data["eventTimestamp"])
+        )
+
+        try:
+            object.__setattr__(
```

Comment:
what?

Reply:
Why are you using `object.setattr` ? 

Oh you're using data class.

Reply:
Is there still a question? This is necessary to set attributes in the `__post_init__` of a frozen dataclass.

Reply:
yeah my bad. Only saw that later.